homeassistant/components/deconz/deconz_event.py

```diff
@@ -0,0 +1,60 @@
+"""Representation of a deCONZ remote."""
+from homeassistant.const import CONF_EVENT, CONF_ID
+from homeassistant.core import EventOrigin, callback
+from homeassistant.helpers.device_registry import CONNECTION_ZIGBEE
+from homeassistant.util import slugify
+
+from .const import _LOGGER, DOMAIN
+
+
+class DeconzEvent:
+    """When you want signals instead of entities.
+
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, device, gateway):
+        """Register callback that will be used for signals."""
+        self._device = device
+        self.gateway = gateway
+
+        self._device.register_async_callback(self.async_update_callback)
+
+        self.event = f"deconz_{CONF_EVENT}"
+        self.id = slugify(self._device.name)
```

Comment:
We shouldn't generate IDs based on names.

Reply:
Why not use `self._device.uniqueid`  ?

Reply:
This is also well established and shouldn't be changed. Using unique id won't be very User friendly

Reply:
but this means that the name changes, the automation breaks ?

Reply:
Unique IDs should not be made for user friendliness, they should be unique and never change